homeassistant/config_entries.py

```diff
@@ -1005,12 +1009,13 @@ async def async_reload(self, entry_id: str) -> bool:
         if (entry := self.async_get_entry(entry_id)) is None:
             raise UnknownEntry
 
-        unload_result = await self.async_unload(entry_id)
+        async with entry.reload_lock:
```

Comment:
I considered swallowing the second reload here but we could legitimately want another reload when one is already in progress.

I suggest that could be done in 2022.7

Reply:
I think this is fine, dropping the second reload just introduces another race which we don't want.